### PR TITLE
JNG-5567 Implementing NavigationExpressionTest from esm-itest

### PR DESCRIPTION
**Pull request by gaborflorian**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-5567" title="JNG-5567" target="_blank">Sub-taskJNG-5567</a>  Implement NavigationExpressionTest from runtime-core-esm-itest in runtime-core-jsl-itest
  </td></table>
  <br />

### CODE REVIEW

File: judo-runtime-core-jsl-itest/models/NavigationTestModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/NavigationTest.java [hunk 1]
```diff
@@ -311,7 +340,475 @@ public void testNavigationPersonWithParents() {
...
+                .withCompA(A1ForCreate.builder().withNumber(4).build())
+                .build()
+        );
+        assertEquals(6, base1Dao.query().selectList()
+                .stream().map(Base1::getRelA)
+                .map(A1::getNumber)
+                .reduce(0, Integer::sum));
+        // TODO
+        //DerivedAttributeCollector derivedAttributeCollector = derivedAttributeCollectorDao.create(DerivedAttributeCollectorForCreate.builder().build());
+        //assertEquals(6, derivedAttributeCollector.getSumRelAonAllBase1());
```
Comment (bencegelei) [inline judo-runtime-core-jsl-itest/models/NavigationTestModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/NavigationTest.java]: Please add JIRA ticket

File: judo-runtime-core-jsl-itest/models/NavigationTestModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/NavigationTest.java [hunk 2]
```diff
@@ -311,7 +340,475 @@ public void testNavigationPersonWithParents() {
...
+                .build()
+        );
+
+        assertEquals(21, base1Dao.query().selectList()
+                .stream().flatMap(b -> b.getBs().stream())
+                .map(B1::getNumber)
+                .reduce(0, Integer::sum));
+        // TODO
+        //DerivedAttributeCollector derivedAttributeCollector = derivedAttributeCollectorDao.create(DerivedAttributeCollectorForCreate.builder().build());
+        //assertEquals(21, derivedAttributeCollector.getSumBsonAllBase1());
```
Comment (bencegelei) [inline judo-runtime-core-jsl-itest/models/NavigationTestModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/NavigationTest.java]: Please add JIRA ticket

File: judo-runtime-core-jsl-itest/models/NavigationTestModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/NavigationTest.java [hunk 3]
```diff
@@ -311,7 +340,475 @@ public void testNavigationPersonWithParents() {
...
+                .withCompA(A1ForCreate.builder().withNumber(4).build())
+                .build()
+        );
+
+        assertEquals(7, base1Dao.querySumRelAonBs(base1).orElseThrow());
+
+    }
+
+    @Test
+    @Disabled()
```
Comment (bencegelei) [inline judo-runtime-core-jsl-itest/models/NavigationTestModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/NavigationTest.java]: MIssing disabled info

File: judo-runtime-core-jsl-itest/models/NavigationTestModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/NavigationTest.java [hunk 4]
```diff
@@ -311,7 +340,475 @@ public void testNavigationPersonWithParents() {
...
+                .stream().map(Base1::getRelA)
+                .map(A1::getNumber)
+                .reduce(0, Integer::sum));
+        // TODO
+        //DerivedAttributeCollector derivedAttributeCollector = derivedAttributeCollectorDao.create(DerivedAttributeCollectorForCreate.builder().build());
+        //assertEquals(6, derivedAttributeCollector.getSumRelAonAllBase1());
+    }
+
+    @Test
+    @Disabled()
```
Comment (bencegelei) [inline judo-runtime-core-jsl-itest/models/NavigationTestModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/NavigationTest.java]: Missing disabled info

**Review by bencegelei — CHANGES_REQUESTED** [review]

File: judo-runtime-core-jsl-itest/models/NavigationTestModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/NavigationTest.java (hunk 4 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/models/NavigationTestModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/NavigationTest.java]: https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/283/commits/b3f1376338026a9b802e7fd8f8e0a38f0f4f2d6c

File: judo-runtime-core-jsl-itest/models/NavigationTestModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/NavigationTest.java (hunk 3 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/models/NavigationTestModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/NavigationTest.java]: https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/283/commits/b3f1376338026a9b802e7fd8f8e0a38f0f4f2d6c

File: judo-runtime-core-jsl-itest/models/NavigationTestModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/NavigationTest.java (hunk 2 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/models/NavigationTestModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/NavigationTest.java]: https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/283/commits/b3f1376338026a9b802e7fd8f8e0a38f0f4f2d6c

File: judo-runtime-core-jsl-itest/models/NavigationTestModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/NavigationTest.java (hunk 1 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/models/NavigationTestModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/NavigationTest.java]: https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/283/commits/b3f1376338026a9b802e7fd8f8e0a38f0f4f2d6c

**Review by robertcsakany — APPROVED** [review]

File: judo-runtime-core-jsl-itest/models/NavigationTestModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/NavigationTest.java [hunk 5]
```diff
@@ -311,7 +340,475 @@ public void testNavigationPersonWithParents() {
...
+                .withCompA(A1ForCreate.builder().withNumber(4).build())
+                .build()
+        );
+        assertEquals(6, base1Dao.query().selectList()
+                .stream().map(Base1::getRelA)
+                .map(A1::getNumber)
+                .reduce(0, Integer::sum));
+        // TODO JNG-5576
+        //DerivedAttributeCollector derivedAttributeCollector = derivedAttributeCollectorDao.create(DerivedAttributeCollectorForCreate.builder().build());
+        //assertEquals(6, derivedAttributeCollector.getSumRelAonAllBase1());
```
Comment (bencegelei) [inline judo-runtime-core-jsl-itest/models/NavigationTestModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/NavigationTest.java]: If the test is already disabled, why is the assert disabled as well?

Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/models/NavigationTestModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/entity/NavigationTest.java]: [47f9ffe](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/283/commits/47f9ffecc74d58a6d6c9154347db7b61c6c70605)

**Review by bencegelei — APPROVED** [review]

**Pull request merged**